apps/server/src/modules/deletion/client/deletion.client.ts

```diff
@@ -22,36 +25,73 @@ export class DeletionClient {
-		const request = this.httpService.post(this.postDeletionRequestsEndpoint, input, this.defaultHeaders());
-
-		return firstValueFrom(request)
-			.then((resp: AxiosResponse<DeletionRequestOutput>) => {
-				// Throw an error if any other status code (other than expected "202 Accepted" is returned).
-				if (resp.status !== 202) {
-					throw new Error(`invalid HTTP status code in a response from the server - ${resp.status} instead of 202`);
-				}
-
-				// Throw an error if server didn't return a requestId in a response (and it is
-				// required as it gives client the reference to the created deletion request).
-				if (!resp.data.requestId) {
-					throw new Error('no valid requestId returned from the server');
-				}
-
-				// Throw an error if server didn't return a deletionPlannedAt timestamp so the user
-				// will not be aware after which date the deletion request's execution will begin.
-				if (!resp.data.deletionPlannedAt) {
-					throw new Error('no valid deletionPlannedAt returned from the server');
-				}
-
-				return resp.data;
-			})
-			.catch((err: Error) => {
-				// Throw an error if sending/processing deletion request by the client failed in any way.
-				throw new Error(`failed to send/process a deletion request: ${err.toString()}`);
-			});
+		let resp: AxiosResponse<DeletionRequestOutput>;
+
+		try {
+			const request = this.httpService.post(this.postDeletionRequestsEndpoint, input, this.defaultHeaders());
+
+			resp = (await firstValueFrom(request)) as AxiosResponse<DeletionRequestOutput>;
+		} catch (err: unknown) {
+			// Throw an error if sending deletion request has failed.
+			throw new BadGatewayException('DeletionClient:queueDeletionRequest', ErrorUtils.createHttpExceptionOptions(err));
+		}
+
+		// Throw an error if any other status code (other than expected "202 Accepted" is returned).
+		if (resp.status !== 202) {
+			const err = new Error(`invalid HTTP status code in a response from the server - ${resp.status} instead of 202`);
+
+			throw new BadGatewayException('DeletionClient:queueDeletionRequest', ErrorUtils.createHttpExceptionOptions(err));
+		}
+
+		// Throw an error if server didn't return a requestId in a response (and it is
+		// required as it gives client the reference to the created deletion request).
+		if (!resp.data.requestId) {
+			const err = new Error('no valid requestId returned from the server');
+
+			throw new BadGatewayException('DeletionClient:queueDeletionRequest', ErrorUtils.createHttpExceptionOptions(err));
+		}
+
+		// Throw an error if server didn't return a deletionPlannedAt timestamp so the user
+		// will not be aware after which date the deletion request's execution will begin.
+		if (!resp.data.deletionPlannedAt) {
+			const err = Error('no valid deletionPlannedAt returned from the server');
+
+			throw new BadGatewayException('DeletionClient:queueDeletionRequest', ErrorUtils.createHttpExceptionOptions(err));
+		}
+
+		return resp.data;
```

Comment:
Please move it all into try after line 37 and throw just Error. Becouse you have there a catch

Reply:
I don't think I understand, what's the point of putting everything into `try`? It's a bit unclear to me what's on your mind here, could you show me some proposal maybe?

Reply:
Note: changed by @SevenWaysDP in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4557/commits/982d83da397174b19def1e3beb3bd94307cd639a , I've also adjusted the other method's implementation to this approach in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4557/commits/61560e91b0e9ed373093bb632ef7516851f53e2c .